### DISCUSSION TITLE
The section that show all the component in a dropdown tree format closes automatically when switched view.

### DISCUSSION THREAD

**Discussion by tanishqkrk** [Atri-Labs/atrilabs-engine#707] [Discuss: DX Improvement Ideas]
I was working with the engine the other day, and was editing some component which was deeply nested in 6 to 7 other components, and I had to switch back and forth between the top component and the one I was working on, and to get to that component, every single time I had to open the main one, then open other component, then another and another until I got to the one I wanted, and after going to another section (like adding a component), the tree closes automatically, so it was a real pain to dig deep in my component tree every time I had to make some changes.
What I suggest is we keep the the previously selected component as an active state and the engine will remember it, so whenever we go back to the tree structure, the engine will already having the active state as the previous one.

**Comment by darshitac11** [comment]
This is a great suggestion, thanks for sharing @tanishqkrk ! 

I usually use the component navigator while working with nested components. But I understand that implementing your suggestion would improve the developer experience - adding it to our roadmap.

**Reply by tanishqkrk** [reply]
My pleasure! I'll keep updating with more DX suggestions.
